tesla_api/__init__.py

```diff
@@ -111,7 +121,9 @@ async def post(self, endpoint, data=None):
         return response_json['response']
 
     async def list_vehicles(self, _class=Vehicle):
-        return [_class(self, vehicle) for vehicle in await self.get('vehicles')]
+        return [_class(self, vehicle) for product in await self.get('vehicles') if 'vehicle_id' in product]
```

Comment:
That may well be true - I couldn't test it (because I have a vehicle and an energy product) so mirrored the logic proposed for the list_energy_sites method